### PR TITLE
Revise focus styles

### PR DESCRIPTION
**Pull request by vladmoroz**
This PR is a pragmatic design workaround to _not_ do https://github.com/modulz/modulz/issues/657

***

Changed from blue focus ring to gray focus ring:
- Button, gray variant
- Select
- Ghost Button, inactive variant
- Toggle Button, inactive variant
- List Item, inactive variant

Added focus styles that were missing:
- Link

Changed from a thin blue focus ring to a thick one:
- Card Link
- Checkbox
- Radio

Demos:
https://radix-cpdhtbbs9.now.sh/docs/popover/
https://radix-cpdhtbbs9.now.sh/docs/ghost-button/

***

Thought process:
- I figured that doing `focus-visible` is going to open a can of worms that we don't want to deal with, no matter whether we are looking at this feature conservatively or not. So I decided to see if this issue may simply be mitigated by adjusting focus styles slightly.
- Initially, I had an idea to add a slight 100-200ms transition for box shadow on `:focus`. I did it and it was terrible. The focus ring just lagged behind mouse clicks, which drew even more attention. Not even sharing those demos 😀
- Since the main offender with mouse interactions was the stark contrast of gray & blue, replacing blue focus ring with gray on gray components solved the issue for me.
  - The update introduces a slight design inconsistency which I personally find helpful — this gets us stronger focus styles for the most CTA-ish elements (e.g. blue buttons) and calmer focus styles for currently inactive and less important elements. Should be fine if we think about focus as an extension of the normal styles.
  - In Modulz, I think this update will greatly reduce how harsh the focused Ghost Buttons look when popovers open and close, yet staying prominent on the screen.

***

Fun fact:
- Green and red buttons have been using green and red focus outlines respectively, but both gray and blue buttons had been using a blue outline. Now this is even and consistent 👌 

***

### CODE REVIEW

**Comment by peduarte** [comment]
Cheers Vlad

I personally preferred the blue focus behaviour, but totally understand the thinking behind it. I think because Radix gets used in a Design Tool these things are much more in-your-face than if it was a website, so I can see why toning it down is a decent solution.

>I figured that doing `focus-visible` is going to open a can of worms that we don't want to deal with
Completely agree

Approved, will wait for @colmtuite approval before merging

**Comment by peduarte** [comment]
@vladmoroz merged `master` here as it contains the new Zeit changes from #199

File: packages/radix/src/components/Checkbox.tsx [hunk 1]
```diff
@@ -18,18 +18,18 @@ export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>((props
...
+            boxShadow: `inset 0 0 0 1px ${theme.colors.gray500}`,
           },
           focus: {
-            borderColor: theme.colors.gray600,
+            boxShadow: `inset 0 0 0 1px ${theme.colors.blue500}, 0 0 0 1px ${theme.colors.blue500}`,
           },
           checked: {
             backgroundColor: theme.colors.blue600,
-            borderColor: theme.colors.blue600,
+            boxShadow: 'none',
```
Comment (benoitgrelard) [inline packages/radix/src/components/Checkbox.tsx]: No focus state at all for checked?

**Comment by vladmoroz** [comment]
@benoitgrelard good point, I did notice the Checkbox and Switch as well. These issues existed before the update, so I didn’t dive into fixing them. Let’s track them as separate bugs?

For the Checkbox specifically, I think we may need to look into how styles config works — `checked` overwrites `focus` in the cascade, so I was unable to fix it in this PR.

File: packages/radix/src/components/Checkbox.tsx (hunk 1 above)
Comment (vladmoroz) [reply packages/radix/src/components/Checkbox.tsx]: No, not at all. This is to have proper styles in normal state at least, without extra border. This didn’t have focus state before as well.

I think we either need to look at precedence of focus and checked states, or have an ability to style checked focus state.

On a related note, I’ve seen separate styles for checked hover and active states many times and I think we should support them as well.

Comment (benoitgrelard) [reply packages/radix/src/components/Checkbox.tsx]: 👍

**Comment by benoitgrelard** [comment]
@vladmoroz yep, makes sense, happy for you to track in a separate issue, I'll approve this one for now then.

**Comment by StephenHaney** [comment]
Interested to see this in the product – it may help reduce the noise factor. The gray color seems to match the gray hover states so it's not two colors changing at once.

I noticed one irregularity, which is tabbing between inputs and selects uses both blue and gray respectively... this struck me as odd:

mixed-blue-gray

**Comment by benoitgrelard** [comment]
@StephenHaney yeah I did find this odd too. Should we switch all the other like input/textarea etc, to gray as well @vladmoroz?

**Comment by vladmoroz** [comment]
Yeah, that Select is included in the popovers specifically to make sure everyone notices it and it doesn’t slip through as a surprise 😀

I did try more grays as a working option — on inputs, text area, checkboxes, etc. Found that it messed with Radix feel too much, and still had to retain the oddity with some other elements. Disliked it strongly.

In the end, figured it’s OK for inputs to have a stronger indicator, since they actively expect keyboard input only.

For comparison, I found during research that Dropbox Paper does focus similarly (either gray or blue outlines depending on component color) and never noticed it actively as a user.

Do you all feel all right about trying this out?

**Comment by benoitgrelard** [comment]
Yeah absolutely we should try it.

**Comment by StephenHaney** [comment]
Works for me!

**Comment by peduarte** [comment]
Just waiting for @colmtuite

**Comment by colmtuite** [comment]
This is great Vlad, thanks!

I also favour the blue outline I think, but very happy to try this and reassess. I like the added consistency on buttons that you mentioned, so I'm on the fence either way.

We will likely have the best focus states on the web soon :)

**Pull request merged**